### PR TITLE
fix: input radio button not changing state

### PR DESCRIPTION
**Pull request by sachdeva-shrey**
Signed-off-by: sachdeva-shrey <shreysachdeva.2000@gmail.com>

Fixes https://github.com/bench-routes/bench-routes/issues/369

## Changes
- Remove redundant code causing the bug.

## Screenshots

https://user-images.githubusercontent.com/47667325/111364682-8f861b80-86b7-11eb-83d4-6b6a08d33e33.mp4

### CODE REVIEW

**Review by muskankhedia — CHANGES_REQUESTED** [review]
@sachdeva-shrey Along with enabling the feature to deselect the fields, you need to remove the data corresponding to that particular field. Based on your changes, the input field is getting hidden for sure but the variables still contain the data which is sent as `body` to the API call. Kindly check [this](https://github.com/bench-routes/bench-routes/blob/master/dashboard/v1.1/src/pages/Input/Input.tsx#L204) API request.

The Input.tsx file `Test and Save` button impact can be noticed in the `local-config.yml` file. You can add all the fields `header`, `params` and `body` for a URL and save it. Now, deselect any one of them and make the API request you will find the deselected data as well in the file.

**Comment by sachdeva-shrey** [comment]
@muskankhedia Thanks for pointing it out, I've fixed this now. Please let me know if it's alright.

**Comment by aquibbaig** [comment]
I am not sure if we'd want users to re-enter the same route with additional parameters just for playing around. I'd propose the following:
- The user fills in the fields describing the URI to be monitored.
- Clicks Test and Save.
- Disable edits to "all" fields and display a message "successfully added a new route, add a new route", when the user clicks on "add a new route" enable edits to all fields.

The point is that we do not want the users to repeatedly play with the tool and add multiple redundant routes to the backend with some minor changes only.

**Comment by sachdeva-shrey** [comment]
Hey @aquibbaig, that's a good point. But, I think it might also increase the hassle for the users who legitimately want to test out with different params. Would it be better to show a confirmation popup, something like "_This route already exists. Are you sure you want to add this route_?" This would allow the users to edit the same params for another request and also add an extra layer to prevent spamming. Please let me know what you think about it.

**Comment by aquibbaig** [comment]
sounds good! @sachdeva-shrey

**Comment by sachdeva-shrey** [comment]
@aquibbaig Thanks! I've made the changes, please review.

https://user-images.githubusercontent.com/47667325/111812558-3fe56100-88fe-11eb-90f9-df124eb99a87.mp4

**Comment by muskankhedia** [comment]
@sachdeva-shrey This looks good. Can you please add some designs to the modal, present one looks bit weird and simple.

**Comment by sachdeva-shrey** [comment]
@muskankhedia Sure, I've added styles to the modal. Please review.

Screenshot from 2021-03-20 13-26-00

File: dashboard/v1.1/src/pages/Input/Input.tsx [hunk 1]
```diff
@@ -104,6 +143,33 @@ const Input = (props: InputScreenProps) => {
...
+
+      try {
+        response = await fetch(`${HOST_IP}/routes-summary`);
+      } catch (err) {
+        console.log(err);
+      }
+
+      const { data } = await response.json();
+      const routes = data.monitoringRoutes;
+      const existingRoutes: string[] = [];
```
Comment (muskankhedia) [inline dashboard/v1.1/src/pages/Input/Input.tsx:159]: Can you please use some other constant name as `existingRoutes` have already been declared in line 125?

File: dashboard/v1.1/src/pages/Input/Input.tsx [hunk 2]
```diff
@@ -104,6 +143,33 @@ const Input = (props: InputScreenProps) => {
...
+      routes.map(route => {
+        let splitURL: string[] = route.split(/(\s+)/);
+        existingRoutes.push(splitURL[2]);
+      });
```
Comment (muskankhedia) [inline dashboard/v1.1/src/pages/Input/Input.tsx:163]:
```suggestion
      routes.forEach(route => {
        let splitURL: string[] = route.split(/(\s+)/);
        existingRoutes.push(splitURL[2]);
      });
```

**Review by muskankhedia — CHANGES_REQUESTED** [review]

File: dashboard/v1.1/src/pages/Input/Input.tsx (hunk 1 above)
Comment (sachdeva-shrey) [reply dashboard/v1.1/src/pages/Input/Input.tsx]: Sure!

**Comment by sachdeva-shrey** [comment]
@muskankhedia I've made the changes, this should be good to go.

**Comment by muskankhedia** [comment]
@sachdeva-shrey Can you please have a look at this, tests are still failing here?

Screenshot 2021-03-21 at 6 17 39 PM

**Comment by sachdeva-shrey** [comment]
@muskankhedia Would it be okay to disable this warning using  `eslint-disable` since fixing this properly might require refactoring a lot of the code here, or should I go ahead and do that?

**Comment by muskankhedia** [comment]
> @muskankhedia Would it be okay to disable this warning using `eslint-disable` since fixing this properly might require refactoring a lot of the code here, or should I go ahead and do that?

yes you can use `eslint-disable` for that.

**Comment by sachdeva-shrey** [comment]
@muskankhedia This is done. Rebased and squashed the commits as well.

**Review by muskankhedia — APPROVED** [review]

**Pull request merged**